src/simplify-type.js

```diff
@@ -44,15 +60,15 @@ function possibleTypes(interfaceType) {
   return interfaceType.possibleTypes.map((t) => t.name);
 }
 
-export default function simplifyType(type) {
+export default function simplifyType(type, whiteList = false) {
   const simplifiedType = {
     name: type.name,
     kind: type.kind
   };
 
   switch (type.kind) {
     case 'OBJECT':
-      simplifiedType.fieldBaseTypes = fieldBaseTypes(type.fields);
+      simplifiedType.fieldBaseTypes = fieldBaseTypes(type.fields, whiteList);
```

Comment:
I started doing that then realized that `fieldBaseTypes` is an object and not an array, which means you can't easily programatically reject keys, this would add a second reduce on `Object.keys(fieldBaseTypes).reduce((acc, field) => { /* filter types */ })` which isn't really any less complex? I agree that it's nicer to limit the concept to only one method, but I'll need to pull out the second reduce to reuse it across to `INTERFACE` and `INPUT_OBJECT`, and then I'll end up with the concept in two methods again. What do you think?